backend/hasura-actions/src/Scheduler.res

```diff
@@ -81,31 +68,74 @@ curl 'http://localhost:5001/api/v1/payments/0xC563388e2e2fdD422166eD5E76971D11eD
 curl 'http://localhost:5001/api/v1/payments/0xC563388e2e2fdD422166eD5E76971D11eD37A466/0x91c0c7b5D42e9B65C8071FbDeC7b1EC54D92AD92' -H 'Content-Type: application/json' --data '{"amount":"100000000000000000","identifier":"86"}'
 */
 
+let getTimestamp = date => {
+  (date->Js.Date.getTime /. 1000.0)->Int.fromFloat
+}
+
+let fromTimeStampToDate = timestamp => {
+  (timestamp->Float.fromInt *. 1000.0)->Js.Date.fromFloat
+}
+
+let getCurrentTimestamp = () => {
+  Js.Date.make()->getTimestamp
+}
+
+let getFinalPayment = (extraPayments: int, maxPayments: int) => {
+  if extraPayments >= maxPayments {
+    maxPayments
+  } else {
+    extraPayments
+  }
+}
+
+/*
+USEFUL INFO
+TokenAddress: 0xC563388e2e2fdD422166eD5E76971D11eD37A466
+RecipientAddress: 0x91c0c7b5D42e9B65C8071FbDeC7b1EC54D92AD92
+Amount = 100000000000000000
+*/
+
 let startProcess = () => {
-  /* let _ = makePayment(
-    ~recipientAddress="0x91c0c7b5D42e9B65C8071FbDeC7b1EC54D92AD92",
-    ~amount="100000000000000000",
-  )*/
+  let now = getCurrentTimestamp()
+  Js.log2("start timestamp:", now)
   let job = CronJob.make(
     #CronString("* * * * *"), // every minute
     _ => {
-      Js.log("Printing every minute")
-      PaymentStreamManager.gqlClient.query(~query=module(Query.GetStreamData), ())
+      Js.log("printing every minute")
+      let currentTimestamp = getCurrentTimestamp()
+      Js.log2("current timestamp:", currentTimestamp)
+      PaymentStreamManager.gqlClient.query(
+        ~query=module(Query.GetStreamData),
+        Query.GetStreamData.makeVariables(~currentTimestamp, ()),
+      )
       ->JsPromise.map(result =>
         switch result {
         | Ok({data: {streams}}) =>
-          Js.log2("wyn success: ", streams)
+          Js.log2("payment streams in focus:", streams)
           let _ = Array.map(streams, stream => {
             let userId = stream.id
             let recipient = stream.recipient
             let amount = stream.amount
-            let paymentTick = stream.paymentTick
+            let nextPayment = stream.nextPayment->Int.toFloat
```

Comment:
Maybe we can abstract the use of float into the functions that need it, and keep most values as integers.

But this is definitely fine as is :)